src/crimson/os/seastore/onode_manager/simple-fltree/onode_node.cc

```diff
@@ -0,0 +1,569 @@
+#include "onode_node.h"
+
+template<size_t BlockSize, int N, ntype_t NodeType>
+auto node_t<BlockSize, N, NodeType>::key_at(unsigned slot) const
+  -> std::pair<const key_t&, const partial_key_t&>
+{
+  auto& key = keys[slot];
+  if constexpr (item_in_key) {
+    return {key, partial_key_t{}};
+  } else {
+    auto p = from_end(key.offset);
+    return {key, *reinterpret_cast<const partial_key_t*>(p)};
+  }
+}
+
+// update an existing oid with the specified item
+template<size_t BlockSize, int N, ntype_t NodeType>
+ghobject_t
+node_t<BlockSize, N, NodeType>::update_oid_with_slot(unsigned slot,
```

Comment:
Every time I see update_oid, I think that we're updating *this, and not the argument.  I don't have a good suggestion for what to replace it with, unfortunately.

Reply:
i renamed it to `get_oid_at()`. hopefully, it's less misleading.

Reply:
Yeah, that works.